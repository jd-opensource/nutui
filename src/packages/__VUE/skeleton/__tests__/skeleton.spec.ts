import { mount } from '@vue/test-utils';
import Skeleton from '../index.vue';

test('should render with rowâ€˜s father content width correctly', () => {
  const wrapper = mount(Skeleton, {
    props: {
      width: '100px'
    }
  });
  const skeleton: any = wrapper.find('.nut-skeleton-content__line');
  expect(skeleton.element.style.width).toBe('100px');
});

test('should allow to disable animation', async () => {
  const wrapper = mount(Skeleton, {
    props: {
      row: '1'
    }
  });

  expect(wrapper.find('.nut-skeleton').exists()).toBeTruthy();

  await wrapper.setProps({ animated: false });
  expect(wrapper.find('.skeleton-animation').exists()).toBeFalsy();
});

test('should change avatar size when using avatarSize prop', () => {
  const wrapper = mount(Skeleton, {
    props: {
      avatar: true,
      avatarSize: '20px'
    }
  });

  const avatar: any = wrapper.find('.avatarClass');
  expect(avatar.element.style.width).toMatchSnapshot('20px');
  expect(avatar.element.style.height).toMatchSnapshot('20px');
});

test('should change avatar shape when using avatarShape prop', () => {
  const wrapper = mount(Skeleton, {
    props: {
      avatar: true,
      avatarShape: 'square'
    }
  });
  expect(wrapper.find('.avatarClass').html()).toMatchSnapshot();
});

test('should be round when using round prop', () => {
  const wrapper = mount(Skeleton, {
    props: {
      title: true,
      round: true,
      avatar: true
    }
  });
  expect(wrapper.find('.avatarClass').exists()).toBeTruthy();
});

test('should render default slot', () => {
  const wrapper = mount(Skeleton, {
    slots: {
      default: 'default'
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});

test('should render correctly when title uses false', () => {
  const wrapper = mount(Skeleton, {
    props: {
      title: false,
      row: '1'
    }
  });

  const content = wrapper.find('.nut-skeleton-content__line');

  expect(content.find('.nut-skeleton-blockTitle').exists()).toBeFalsy();
  expect(content.findAll('.nut-skeleton-blockLine').length).toBe(1);
});
