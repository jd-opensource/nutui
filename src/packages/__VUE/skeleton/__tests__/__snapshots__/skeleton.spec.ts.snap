// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should change avatar shape when using avatarShape prop 1`] = `"<view style="width: 50px; height: 50px; background-color: rgb(238, 238, 238); color: rgb(102, 102, 102);" class="nut-avatar nut-avatar-normal nut-avatar-square avatarClass avatarClass--square"></view>"`;

exports[`should change avatar size when using avatarSize prop > 20px 1`] = `"20px"`;

exports[`should change avatar size when using avatarSize prop > 20px 2`] = `"20px"`;

exports[`should render default slot 1`] = `
"<view class="nut-skeleton">
  <!--v-if-->
  <view class="nut-skeleton-content">
    <!--v-if-->
    <view class="nut-skeleton-content__line" style="width: 100px;">
      <view class="nut-skeleton-blockTitle" style="height: 15px;"></view>
      <view class="nut-skeleton-blockLine" style="height: 15px;"></view>
    </view>
  </view>
</view>"
`;
